### PR TITLE
CEM-2045 box composition

### PR DESCRIPTION
**Pull request by roggc**
do two squares and a rectangle with images that changes coinciding with animation
agorman
as you can see the animation of the text and the squares do not coincide in time.

### CODE REVIEW

File: src/Containers/BoxComposition/BoxComposition.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import styled from 'styled-components'
+
+interface IBoxProps{
```
Comment (ralph-dev) [inline src/Containers/BoxComposition/BoxComposition.tsx:4]: Why don't you extend ImageProps???

**Review by ralph-dev — CHANGES_REQUESTED** [review]
You need to pass ...props to all new components / make sure you are extending the correct interface.

File: src/Containers/BoxComposition/BoxComposition.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { Meta, Story } from '@storybook/react';
+import { BoxComposition, IBoxCompositionProps } from '@Containers';
+import { createStoryTitle } from '../../Constants';
+
+export default {
+    title: createStoryTitle('BoxComposition'),
+    component: BoxComposition,
+    args: {
+        data:[
+            {
```
Comment (ralph-dev) [inline src/Containers/BoxComposition/BoxComposition.stories.tsx:11]: This structure won't work, 3 images are part of one set :) try to actually replace the content on the pulse then you will see why :)

File: src/Containers/BoxComposition/BoxComposition.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import styled from 'styled-components'
+import {SmallText} from '@Text'
+import {Mixins} from '@Utils'
+
+const ANIMATION_DURATION=7
+
+interface IBoxProps extends IImgProps,IDivProps{
+    imgSource:string;
+    withText:boolean;
```
Comment (ralph-dev) [inline src/Containers/BoxComposition/BoxComposition.tsx:10]: Why do you need this?
Just do
!!text &&

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Code structure and props need work, almost there

File: src/Containers/BoxComposition/BoxComposition.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    text:string;
+    id:string;
+}
+
+const Box:React.FC<IBoxProps>=({imgSource,top,left,width,height,withText,text}):React.ReactElement=>
```
Comment (Ampatte2) [inline src/Containers/BoxComposition/BoxComposition.tsx:15]: Why is this a FC and not a function called renderBox = (props) => jsx?

File: src/Containers/BoxComposition/BoxComposition.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    `}
+    border-radius:5px;
+    box-shadow:0 0 1px 0 black;
+`
+
+export interface IBoxCompositionProps{
+    data:IBoxProps[];
+}
+
+export const BoxComposition:React.FC<IBoxCompositionProps>=({data}):React.ReactElement=>
```
Comment (Ampatte2) [inline src/Containers/BoxComposition/BoxComposition.tsx:76]: A FC that just maps? Is there additional styling to be included for this? If not I'd rather just export the Box component and for the story map it out for display.

File: src/Containers/BoxComposition/BoxComposition.stories.tsx (hunk 2 above)
Comment (ralph-dev) [reply src/Containers/BoxComposition/BoxComposition.stories.tsx:11]: [{
mainImage: {imgSources: 'stringUrl',
                top:0,
                left:75,
                width:100,
                height:200,
                text:'bla bla bla',
                id:'0'},
leftImage: {},
rightImage: {}
}, {Set2}]

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Wrong, no need for two arrays.
[
{MainImage: {}, LeftImage: {}, RightImage: {}},
{MainImage: {}, LeftImage: {}, RightImage: {}}
]
Look above ^

File: src/Containers/BoxComposition/BoxComposition.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import { BoxComposition, IBoxCompositionProps } from '@Containers';
+import { createStoryTitle } from '../../Constants';
+
+export default {
+    title: createStoryTitle('BoxComposition'),
+    component: BoxComposition,
+    args: {
+        data:[{
+            main:{
+                imgSource:['https://dynaimage.cdn.cnn.com/cnn/c_fill,g_auto,w_1200,h_675,ar_16:9/https%3A%2F%2Fcdn.cnn.com%2Fcnnnext%2Fdam%2Fassets%2F210204090136-amanda-gorman-time-cover.jpg'],
```
Comment (ralph-dev) [inline src/Containers/BoxComposition/BoxComposition.stories.tsx:12]: imgSource: string
no need for 3 arrays

**Review by ralph-dev — CHANGES_REQUESTED** [review]
All imgSource should be strings no need for the array of strings

**Review by ralph-dev — APPROVED** [review]
LGTM, make sure the component still works before merging as the structure was modified.

**Pull request merged**